### PR TITLE
SAR-10026 Move to hmrc-mongo

### PR DESCRIPTION
**Pull request by EviNikolaidou**

### CODE REVIEW

File: app/uk/gov/hmrc/soletraderidentificationfrontend/repositories/JourneyConfigRepository.scala [hunk 1]
```diff
@@ -17,78 +17,85 @@
...
+      Filters.and(
+        Filters.equal(JourneyIdKey, journeyId),
+        Filters.equal(AuthInternalIdKey, authInternalId)
+      )
+    ).headOption
+
   }
 
-  setIndex()
+  def count: Future[Long] = collection.countDocuments().toFuture()
```
Comment (HarrNM) [inline app/uk/gov/hmrc/soletraderidentificationfrontend/repositories/JourneyConfigRepository.scala:67]: Is this used anywhere?

File: app/uk/gov/hmrc/soletraderidentificationfrontend/repositories/JourneyConfigRepository.scala [hunk 2]
```diff
@@ -17,78 +17,85 @@
...
   val AuthInternalIdKey = "authInternalId"
   val CreationTimestampKey = "creationTimestamp"
 
   implicit val journeyConfigMongoFormat: OFormat[JourneyConfig] = Json.format[JourneyConfig]
+
+  def timeToLiveIndex(timeToLiveDuration: Long): IndexModel = {
+    IndexModel(
+      keys = ascending(CreationTimestampKey),
+      indexOptions = IndexOptions()
+        .name("PartnershipInformationExpires")
```
Comment (HarrNM) [inline app/uk/gov/hmrc/soletraderidentificationfrontend/repositories/JourneyConfigRepository.scala:96]: Maybe rename to SoleTraderInformationExpires

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -17,19 +17,23 @@ object AppDependencies {
...
     "org.mockito" % "mockito-core" % "3.11.2" % Test,
-    "org.scalatestplus" %% "mockito-3-4" % "3.2.9.0" % Test
+    "org.scalatestplus" %% "mockito-3-4" % "3.2.9.0" % Test,
+    "uk.gov.hmrc.mongo" %% "hmrc-mongo-test-play-28" % "0.67.0" % Test
   ) ++ commonTestDependencies(Test)
 
   val it: Seq[ModuleID] = Seq(
-    "com.github.tomakehurst" % "wiremock-jre8" % "2.29.1" % IntegrationTest
+    "com.github.tomakehurst" % "wiremock-jre8" % "2.29.1" % IntegrationTest,
+    "uk.gov.hmrc.mongo" %% "hmrc-mongo-test-play-28" % "0.67.0" % IntegrationTest
```
Comment (HarrNM) [inline project/AppDependencies.scala:36]: Do we need this in all 3 Sequences, or does it work in just the commonTestDependencies

File: app/uk/gov/hmrc/soletraderidentificationfrontend/repositories/JourneyConfigRepository.scala (hunk 1 above)
Comment (EviNikolaidou) [reply app/uk/gov/hmrc/soletraderidentificationfrontend/repositories/JourneyConfigRepository.scala:67]: Its used in the tests

File: project/AppDependencies.scala (hunk 3 above)
Comment (HarrNM) [reply project/AppDependencies.scala:35]: Realise now it's two different dependencies, so ignore me
"uk.gov.hmrc.mongo" %% "hmrc-mongo-play-28" % "0.67.0" and
"uk.gov.hmrc.mongo" %% "hmrc-mongo-test-play-28" % "0.67.0"

**Review by HarrNM — APPROVED** [review]

**Pull request merged**